src/main/scala/mesosphere/marathon/state/Instance.scala

```diff
@@ -0,0 +1,80 @@
+package mesosphere.marathon
+package state
+
+import core.instance.{Reservation, Instance => CoreInstance}
+import mesosphere.marathon.core.condition.Condition
+import mesosphere.marathon.core.instance.Instance.InstanceState
+import mesosphere.marathon.core.task.Task
+import mesosphere.marathon.raml.Raml
+import play.api.libs.functional.syntax._
+import play.api.libs.json.Reads._
+import play.api.libs.json._
+
+case class Instance(
```

Comment:
why do we have to have two anyway? Why can't we just when serializing "core instance" say that we use only the version of runspec when creating the protobuf?

Reply:
Serializing is easy but deserializing is not. How would you deserialize to the core instance if you do not have the run spec? I'm happy for any suggestion. The state instance helps to handle migrations without loading the run spec. It is simply a lean storage model.